src/ica_api.c

```diff
@@ -2759,8 +2767,16 @@ unsigned int ica_aes_ctr(const unsigned char *in_data, unsigned char *out_data,
 	unsigned int function_code;
 
 #ifdef ICA_FIPS
+	/* Note that the FIPS ctr check cannot detect ctr wraps
+	 * over chained calls to this function. */
+	unsigned long num_blocks = data_length / DES_BLOCK_SIZE;
+	unsigned int num_additional_bytes = data_length % DES_BLOCK_SIZE;
 	if (fips >> 1)
 		return EACCES;
+	if (num_additional_bytes > 0)
+		num_blocks++;
+	if (ctr_width < 64U && num_blocks > (1ULL << ctr_width))
```

Comment:
num_blocks >= (1ULL << ctr_width) ?

Reply:
num_blocks == 2^ctr_width works. I just tested it again.

Reply:
ah yeah i see. "off by one" in my head